package/MDAnalysis/analysis/nucleicacids.py

```diff
@@ -140,6 +148,41 @@ def __init__(self, selection1: List[mda.AtomGroup],
             self._s1 += selection1[i]
             self._s2 += selection2[i]
 
+    @staticmethod
+    def select_strand_atoms(
+        strand1: List[Residue], strand2: List[Residue], 
+        a1_name: str, a2_name: str, guanine_name: str = 'G',
+        adenine_name: str = 'A', uracil_name: str = 'U',
+        tymine_name: str = 'T', cytosine_name: str = 'C'
+        ) -> Tuple[List[mda.AtomGroup], List[mda.AtomGroup]]:
+        pyrimidines: List[str] = [cytosine_name, tymine_name, uracil_name]
+        purines: List[str] = [adenine_name, guanine_name]
+        
+        sel1: List[mda.AtomGroup] = []
+        sel2: List[mda.AtomGroup] = []
+        strand: Tuple[Residue, Residue] = zip(strand1, strand2)
```

Comment:
the type annotations are a great addition.  I think this one (from `zip`) is technically a generator/iterator.  I think instead you want to annotate `s: Tuple[Residue, Residue]`

Reply:
I just moved zip into the for look line.